src/YAMLValidator/InputProcessor.cs

```diff
@@ -0,0 +1,110 @@
+
+        var pathOption = new Option<string>(
+            name: "--path",
+            description: "The path to the input yaml file or directory of yaml files"
+        )
+        { IsRequired = true };
+
+        pathOption.AddValidator(result =>
+        {
+            var inputFilePath = result.GetValueForOption(pathOption);
+
+            // either file or folder must be passed
+            var pathType = string.Empty;
+            if (string.IsNullOrEmpty(inputFilePath))
+            {
+                result.ErrorMessage = "The input is invalid, input must be a filepath to a yaml file \\" +
+                "or a folder path to a folder of yaml files";
+            }
+            else if (!Directory.Exists(inputFilePath) && !File.Exists(inputFilePath))
+            {
+                result.ErrorMessage = "The input path does not exist";
+            }
+            else if (Directory.Exists(inputFilePath))
+            {
+                if (Directory.GetFiles(inputFilePath, $"*{Constants.YamlFileExtension}").Length == 0)
+                {
+                    result.ErrorMessage = "The input folder does not contain any yaml files";
+                }
+            }
+            else if (File.Exists(inputFilePath))
+            {
+                if (Path.GetExtension(inputFilePath) != Constants.YamlFileExtension)
+                {
+                    result.ErrorMessage = "The input file must be a yaml file";
+                }
+            }
+        });
+
+        // assume local schema file exists in nuget package, use relative filepath for now
+        var schemaOption = new Option<string>(
+            name: "--schema",
+            description: "The path to the schema json file",
+            getDefaultValue: () => @".\schema\pa.yaml-schema.json"
+            );
+
+        schemaOption.AddValidator(result =>
+        {
+            var schemaPath = result.GetValueForOption(schemaOption);
+            if (string.IsNullOrEmpty(schemaPath))
+            {
+                result.ErrorMessage = "Schema option selected, but no schema was provided";
+            }
+            else if (Path.GetExtension(schemaPath) != Constants.JsonFileExtension)
+            {
+                result.ErrorMessage = "The schema file must be a json file";
+            }
+            else if (!File.Exists(schemaPath))
+            {
+                result.ErrorMessage = "The schema file does not exist";
+            }
+        });
+
+        // define root
+        var rootCommand = new RootCommand("YAML validator cli-tool");
+
+        // validate command
+        var validateCommand = new Command("validate", "Validate the input yaml file")
+        {
+            pathOption,
+            schemaOption
+        };
+
+        validateCommand.SetHandler((pathOptionVal, schemaOptionVal) =>
+        {
+            // validation has completed, we either have a file or folder
+            var pathType = File.Exists(pathOptionVal) ? Constants.FileTypeName :
```

Comment:
This is a more explicit way to validate if a path leads to a folder:
`File.GetAttributes(filePath).HasFlag(FileAttributes.Directory);`

Reply:
added and fixed this